modules/world/evaluation/labels/base_label_function.hpp

```diff
@@ -0,0 +1,48 @@
+// Copyright (c) 2020 fortiss GmbH
+//
+// This work is licensed under the terms of the MIT license.
+// For a copy, see <https://opensource.org/licenses/MIT>.
+
+#ifndef MODULES_WORLD_EVALUATION_LABELS_BASE_LABEL_FUNCTION_HPP_
+#define MODULES_WORLD_EVALUATION_LABELS_BASE_LABEL_FUNCTION_HPP_
+
+#include <memory>
+#include <string>
+#include <unordered_map>
+#include <utility>
+#include <vector>
+
+#include "ltl/label.h"
+#include "ltl/rule_monitor.h"
+#include "modules/world/objects/object.hpp"
+
+namespace modules {
+namespace world {
+
+class ObservedWorld;
+
+namespace evaluation {
+using ltl::Label;
+using LabelMap = ltl::EvaluationMap;
+
+class BaseLabelFunction {
+ public:
+  explicit BaseLabelFunction(std::string label_str)
+      : label_str_(std::move(label_str)) {}
+  virtual std::vector<LabelMap::value_type> Evaluate(
+      const world::ObservedWorld &observed_world) const = 0;
+  const std::string &GetLabelStr() const { return label_str_; }
```

Comment:
reformat so std::string& everywhere

Reply:
I suggest to change

https://github.com/bark-simulator/bark/blob/cc393384134188bf68d098a9efa486ff36069577/.clang-format#L54

to `DerivePointerAlignment: false` then.

Reply:
I changed it to false